File: examples/raw-nodejs/webhook-verify.js
'use strict';

const { verifyWebhook } = require('./payxcommerce');

const webhookSecret = 'YOUR_WEBHOOK_SECRET';
let rawBody = '';

process.stdin.setEncoding('utf8');
process.stdin.on('data', (chunk) => { rawBody += chunk; });
process.stdin.on('end', () => {
  try {
    const payload = verifyWebhook(rawBody, {
      'X-PXC-Event-ID': process.env.HTTP_X_PXC_EVENT_ID || '',
      'X-PXC-Timestamp': process.env.HTTP_X_PXC_TIMESTAMP || '',
      'X-PXC-Signature': process.env.HTTP_X_PXC_SIGNATURE || ''
    }, webhookSecret);

    // Store event_id and skip duplicates before updating local order status.
    // Map event_type to your local order lifecycle.
    console.log('PayXCommerce webhook verified:', payload.event_type || 'unknown');
  } catch (error) {
    console.error('Invalid webhook:', error.message);
    process.exit(1);
  }
});

File: examples/raw-python/webhook_verify.py
import os
import sys
from payxcommerce import verify_webhook

WEBHOOK_SECRET = "YOUR_WEBHOOK_SECRET"

raw_body = sys.stdin.read()
headers = {
    "X-PXC-Event-ID": os.environ.get("HTTP_X_PXC_EVENT_ID", ""),
    "X-PXC-Timestamp": os.environ.get("HTTP_X_PXC_TIMESTAMP", ""),
    "X-PXC-Signature": os.environ.get("HTTP_X_PXC_SIGNATURE", ""),
}

try:
    payload = verify_webhook(raw_body, headers, WEBHOOK_SECRET)
except Exception as exc:
    print(f"Invalid webhook: {exc}")
    raise SystemExit(1)

# Store event_id and skip duplicates before updating local order status.
# Map event_type to your local order lifecycle.
print("PayXCommerce webhook verified:", payload.get("event_type", "unknown"))
